physlr/physlr.py

```diff
@@ -1332,6 +1332,79 @@ def determine_molecules_bc_cosine_of_squared(g, u):
                 for community in
                 Physlr.detect_communities_cosine_of_squared(g, bi_connected_component)]
 
+    @staticmethod
+    def partition_subgraph_into_bins_randomly(node_set, max_size=40):
+        """Prtition the subgraph into bins randomly for faster processing. Return bins."""
+        bins_count = 1
+        if len(node_set) > max_size:
+            bins_count = 1 + int(len(node_set) / max_size)
+        node_list = list(node_set)
+        random.shuffle(node_list)
+        size, leftover = divmod(len(node_set), bins_count)
+        bins = [node_list[0 + size * i: size * (i + 1)] for i in list(range(bins_count))]
+        edge = size * bins_count
+        for i in list(range(leftover)):
+            bins[i % bins_count].append(node_list[edge + i])
+        bin_sets = [set() for _ in range(len(bins))]
+        for i, c in zip(range(len(bins)), bins):
+            bin_sets[i].update(set(c))
+        return bin_sets
+
+    @staticmethod
+    def merge_communities(g, communities, node_set=0, strategy=0, cutoff=20):
+        """Merge communities if appropriate."""
+        mode = 1
+        if cutoff == -1:  # no merging
+            return communities
+        if len(communities) == 1 and (node_set == 0 or strategy != 1):
+            return communities
+        if strategy == 1:  # Merge by Initializing Louvain with the communities
+            return Physlr.detect_communities_louvain(g, node_set, communities)
+        # Ad-hoc Merge (default - strategy = 0)
+        merge_network = nx.Graph()
+        for i in range(len(communities)):
+            merge_network.add_node(i)
+        for i, com1 in enumerate(communities):
+            for k, com2 in enumerate(communities):
```

Comment:
```suggestion
            for j, com2 in enumerate(communities):
```
Use `j` if possible.

Reply:
At first I set to `j` and `pylint` ouput error as I remember :D I'll check it again.

Reply:
`physlr/physlr.py:1370:16: C0103: Variable name "j" doesn't conform to snake_case naming style (invalid-name)` pylints error.

Reply:
That's weird! a single-character word `j` does not conform to snake_case naming style while `i` and 'k' does!

Reply:
and there is another j used in another loop just after this loop, and pylint has not problem with that one :laughing:

Reply:
Just add `j` to `good-names` in `.pylintrc`. =)

Reply:
:+1: